pylds/base.py

```diff
@@ -6,32 +6,122 @@
 Publisher, Number(Volume No), pp.142-161.
 """
 
+import sys
 import numpy as np
-import matplotlib.pyplot as plt
+import numpy.ma as ma
+from operator import itemgetter
 from scipy.integrate import solve_ivp
+from functools import reduce
+
+def energy_conservation_condition(phase_space_axes, H0, potential_energy, momentum_sign):
+    N_dim = len(phase_space_axes)
+    phase_space_positions = np.transpose(phase_space_axes[:int(N_dim/2)])
+    phase_space_momenta = np.transpose(phase_space_axes[int(N_dim/2):])
+    V = potential_energy(phase_space_positions)
+    points_dof_remaining = momentum_sign * \
+        np.sqrt(2*(H0 - V) - (phase_space_momenta**2).sum(axis=1))
+    return points_dof_remaining
 
 def generate_points(grid_parameters):
     """
     Returns a 1D array of all points from a on a uniform grid with dimensions and size defined by list of input parameters.
     An additional dimension initiallised with zeros is added for the calculation of Lagrangian Descriptors.
-    
+
     Parameters
     ----------
-    grid_parameters : list of 3-tuples of floats
+    slice_parameters : list of 3-tuples of floats
```

Comment:
list or dictionary

Reply:
Need to think about how to write that description. Thanks!

Reply:
Updated!